airmozilla/main/views.py

```diff
@@ -1114,6 +1115,93 @@ def item_pubdate(self, event):
         return event.start_time
 
 
+def related_content(request, slug):
+    event = get_object_or_404(Event, slug=slug)
+
+    es = pyelasticsearch.ElasticSearch(settings.RELATED_CONTENT_URL)
+    mlt_query = {
+        "more_like_this": {
+            "fields": ["tags", "title", "channels"],
+            "docs": [
+                {
+                    "_index": "events",
+                    "_type": "event",
+                    "_id": event.id
+                }],
+            "min_term_freq": 1,
+            "max_query_terms": 5,
+        }
+    }
+
+    if request.user.is_active:
+        if is_contributor(request.user):
+            query = {
+                "query": {
+                    "filtered": {
+                        "query": mlt_query,
+                        "filter": {
+                            "must_not": {
+                                "term": {
+                                    "privacy": Event.PRIVACY_COMPANY
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        else:
+            query = {
+                "query": mlt_query
+            }
+    else:
+        query = {
+            "query": {
+                "filtered": {
+                    "query": mlt_query,
+                    "filter": {
+                        "bool": {
+                            "must": {
+                                "term": {"privacy": Event.PRIVACY_PUBLIC}
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+    ids = []
+    hits = es.search(query, index='events')['hits']
+
+    for doc in hits['hits']:
+        print "\t", repr(doc['_source']['title']), doc['_id']
+        ids.append(int(doc['_id']))
+
+    if request.user.is_active:
+        if is_contributor(request.user):
+            events = Event.objects \
+                          .filter(id__in=ids)
+        else:
+            events = Event.objects.scheduled_or_processing()
+    else:
+        events = Event.objects \
+                      .filter(id__in=ids)
+
+    curated_groups_map = collections.defaultdict(list)
+    events_list = list(events)
+    events_sorted = sorted(events_list, key=lambda e: ids.index(e.id))
```

Comment:
You can simplify these two lines with:
```python
events = sorted(events, key=lambda e: ids.index(e.id))
```
(Note you'll need to change line 1197 too.